### PR TITLE
Test for zombie processes in crashtracking

### PR DESCRIPTION
**Pull request by kevingosse**
## Motivation
https://datadoghq.atlassian.net/browse/APMLP-289 and https://datadoghq.atlassian.net/browse/APMLP-290

## Changes

Adds an injection test validating that crashtracking:
 - does not spawn a child process unless there's a crash
 - does not leave zombie processes behind

To look for child processes, the weblog exposes a `/child_pids` endpoint that browses the `/proc` folder to find the parent processes. It doesn't rely on `ps` because it's not available in some containers.
To look for zombie processes, the weblog exposes a `/fork_and_crash` that spawns a copy of the process (using fork whenever possible, or just manually starting a process on Java and .NET) then have it crash. We need a child process for this because if we crash the main process then the container will be teared down and we won't be able to observe any zombie.

The tests currently fail on Ruby, I think it runs without the crashtracking fix.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/auto_inject/utils.py [hunk 1]
```diff
@@ -37,14 +39,31 @@ def _test_install(self, virtual_machine, profile: bool = False):
             logger.info(f"Http request done with uuid: [{request_uuid}] for ip [{vm_ip}]")
         wait_backend_trace_id(request_uuid, 120.0, profile=profile)
 
+    def warmup(self, virtual_machine):
+        vm_ip = virtual_machine.get_ip()
+        vm_port = virtual_machine.deffault_open_port
+        warmup_weblog(f"http://{vm_ip}:{vm_port}/")
+
+    def get_child_pids(self, virtual_machine) -> str:
```
Comment (robertomonteromiguel) [inline tests/auto_inject/utils.py:47]: why not add this methods to the weblog interface?

File: utils/_context/_scenarios/__init__.py [hunk 2]
```diff
@@ -651,6 +651,14 @@ def all_endtoend_scenarios(test_object):
         github_workflow="libinjection",
     )
 
+    container_auto_injection_install_script_crashtracking = InstallerAutoInjectionScenario(
```
Comment (robertomonteromiguel) [inline utils/_context/_scenarios/__init__.py:654]: I advocate for using the existing scenario "INSTALLER_AUTO_INJECTION" . And after break the app, restore the app

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
- This PR is very large, I would have preferred to have divided it into several parts, for example, by language.
- There are lint failures
- There are failures on gitlab
- I propose not to create a new scenario, but to use the existing scenario “INSTALLER_AUTO_INJECTION”. This new test case will be executed together with other tests. The condition is after breaking the application, restore it.

File: utils/_context/_scenarios/__init__.py (hunk 2 above)
Comment (kevingosse) [reply utils/_context/_scenarios/__init__.py]: Done

File: tests/auto_inject/utils.py (hunk 1 above)
Comment (kevingosse) [reply tests/auto_inject/utils.py]: Done

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
Only a minimal change request

File: tests/auto_inject/test_auto_inject_install.py [hunk 3]
```diff
@@ -75,6 +77,61 @@ def test_profiling(self, virtual_machine):
...
+@scenarios.installer_auto_injection
+class TestContainerAutoInjectInstallScriptCrashTracking_NoZombieProcess(base.AutoInjectBaseTest):
+    @parametrize_virtual_machines(
+        bugs=[
+            {"library": "ruby", "reason": "APMLP-312"},
+            {"weblog_variant": "test-app-java-buildpack", "reason": "APMON-1595"},
+            {"weblog_variant": "test-app-python-alpine", "reason": "APMLP-290"},
+        ]
+    )
+    def test_install(self, virtual_machine):
```
[the remark this answers is not in the document]
Comment (kevingosse) [reply tests/auto_inject/test_auto_inject_install.py]: Good call, fixed in https://github.com/DataDog/system-tests/pull/3364/commits/dd455a0766361d8b04fc8dcc8bd061df2b089d47

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**